cernopendata_client/search.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# This file is part of cernopendata-client.
+#
+# Copyright (C) 2020 CERN.
+#
+# cernopendata-client is free software; you can redistribute it and/or modify
+# it under the terms of the GPLv3 license; see LICENSE file for more details.
+
+from __future__ import print_function
+
+import argparse
+import os
+import sys
+
+import click
+import requests
+
+try:
+    from urllib.parse import quote
+except ImportError:
+    # fallback for Python 2
+    from urllib import quote
+
+
+def verify_recid(recid=None, server=None):
+    """Verify that recid corresponds to a valid Open Data record webpage."""
+    if recid is None:
+        sys.exit(
+            "ERROR: You must supply a record id number as an " "input using -r flag."
+        )
+    else:
+        input_record_url = server + "/record/" + str(recid)
+        input_record_url_check = requests.get(input_record_url)
+
+        if input_record_url_check.status_code == 200:
+            base_record_id = str(recid)
+            return base_record_id
+        else:
+            try:
+                input_record_url_check.raise_for_status()
+            except requests.HTTPError as http_error_msg:
+                print("ERROR: The record id number you supplied is not valid.")
+                sys.exit(http_error_msg)
+            return False
+
+
+def get_recid_api(base_record_id=None, server=None):
```

Comment:
`server` first, as elsewhere.

Please also think of providing some validation.  Currently one gets:

```console
$ cernopendata-client get-file-locations --recid 15007 --protocol http --server something 
...
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/requests/models.py", line 387, in prepare_url
    raise MissingSchema(error)
requests.exceptions.MissingSchema: Invalid URL 'something/record/15007': No schema supplied. Perhaps you meant http://something/record/15007?
```

This is good, but seeing Python traceback may be confusing for some users.

Perhaps validate server arguments after they are passed by:
- adding a validator function in `validator.py`
- calling it right after click entry points
- the validator could for now simply check whether the argument starts with `http://...` and return true/false
- in case of wrong argument, we can fail gracefully with an error message, not showing python traceback